### PR TITLE
DEPR: rename `startingMonth` to `starting_month` (argument in BQuarterBegin)

### LINKED ISSUE #57794: DEPR: rename `startingMonth` to `starting_month` (argument in BQuarterBegin)
Renamings should be done with care, but this one strikes me as especially odd

```python
pandas.tseries.offsets.QuarterBegin(startingMonth=1)
```
It looks very odd in Python to have a camelcase argument name...I thought this was probably a typo in the docs when I saw it, but no, it runs

OK with deprecating in favour of `starting_month`?

I think this is the only place in pandas where this happens:
```console
$ git grep -E ' [a-z]+[A-Z][a-z]+: ' pandas
pandas/_libs/tslibs/offsets.pyi:        self, n: int = ..., normalize: bool = ..., startingMonth: int | None = ...
pandas/_libs/tslibs/offsets.pyi:        startingMonth: int = ...,
pandas/_libs/tslibs/offsets.pyi:        startingMonth: int = ...,
$ git grep -E ' [a-z]+[A-Z][a-z]+ : ' pandas
pandas/_libs/tslibs/offsets.pyx:    startingMonth : int, default 3
pandas/_libs/tslibs/offsets.pyx:    startingMonth : int, default 3
pandas/_libs/tslibs/offsets.pyx:    startingMonth : int, default 3
pandas/_libs/tslibs/offsets.pyx:    startingMonth : int, default 3
pandas/_libs/tslibs/offsets.pyx:    startingMonth : int {1, 2, ... 12}, default 1
pandas/_libs/tslibs/offsets.pyx:    startingMonth : int {1, 2, ..., 12}, default 1
```

### PR DESCRIPTION
**Pull request by gboeker**
- [x] closes #57794
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by Aloqeely** [comment]
You need to deprecate `startingMonth` first

**Comment by gboeker** [comment]
Thanks @Aloqeely ! Could you explain to me how I can deprecate `startingMonth`?

**Comment by Aloqeely** [comment]
Basically you should still allow `startingMonth`, but if it is used you should warn the user that it will be removed soon
I think a check in `__init__` will do the job but I'm not sure, cc @jbrockmendel

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -2657,49 +2657,49 @@ cdef class YearBegin(YearOffset):
...
     # FIXME(cython#4446): python annotation here gives compile-time errors
     # _default_starting_month: int
     # _from_name_starting_month: int
 
     cdef readonly:
-        int startingMonth
+        int starting_month
 
-    def __init__(self, n=1, normalize=False, startingMonth=None):
+    def __init__(self, n=1, normalize=False, starting_month=None):
```
[the remark this answers is not in the document]
Comment (gboeker) [reply pandas/_libs/tslibs/offsets.pyx:2670]: Hi Brock, I changed it. Could you take a look?

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**